java/src/main/java/ai/rapids/cudf/CSVWriterOptions.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *
+ *  Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class CSVWriterOptions {
```

Comment:
If this writer will be supporting nested types, which I assume it will at some point, does it not make sense to extend the `ColumnWriterOptions` just like we do for `ParquetWriterOptions` and `OrcWriterOptions`?

Reply:
I'll see about extending `ColumnWriterOptions`, but I don't currently see CSV supporting nested types.

Reply:
Hmm. The more I look at this, the more it looks like this doesn't fit.

`CSVWriterOptions` are the options for writing the entire table to CSV, while `ColumnWriterOptions` seems to be per-column. `StructColumnWriterOptions` seems like a closer fit.
Then, again, extending `StructColumnWriterOptions` opens the door to users calling `withStructColumn`, `withListColumn`, etc. We would then be trading compile-time errors with runtime assertions.

I am inclined to revisit this when nested columns are attempted for CSV.